### PR TITLE
[feat] Implement GitHub Integration System with Webhook Support

### PR DESCRIPTION
**Pull request by toyamarinyon**
https://github.com/user-attachments/assets/c291a110-4729-47dc-ad0f-3019f01fff83

## Changes
- Add GitHub integration system with unique ID support (`gthb_` prefix)
- Implement webhook handler for flow execution based on GitHub events
- Create persistent settings storage for GitHub configurations
- Add flow path detection with `allFlowEdges` utility
- Implement form handling for GitHub integration setup
- Update repository selection to use full repository names
- Add type definitions for GitHub webhook events and actions
- Normalize webhook payload parsing for consistent line endings

## Why
This implementation provides a complete GitHub integration system that allows users to:
1. Configure and persist GitHub webhook integrations
2. Connect specific repository events to agent flows
3. Execute flows based on GitHub events with proper webhook handling
4. Maintain configuration state across sessions
5. Better manage flow paths between action nodes

The changes improve type safety, user experience, and provide a robust foundation for GitHub-based automation workflows.

## Testing
- Verified webhook payload parsing with normalized line endings
- Tested flow path detection with `allFlowEdges` utility
- Validate

### CODE REVIEW

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+		where: (agents, { eq }) => eq(agents.id, input.agentId),
+	});
+	if (agent === undefined) {
+		throw new Error(`Agent with id ${input.agentId} not found`);
+	}
+	await db
+		.insert(gitHubIntegrations)
+		.values({
+			id: input.id ?? generateId(),
+			dbId: 123,
```
Comment (shige) [inline app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts]: question: Will this code be subject to changes in the future?

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+			callSign: input.callSign,
+			event: input.event,
+			startNodeId: input.startNodeId,
+			endNodeId: input.endNodeId,
+			nextAction: input.nextAction,
+		})
+		.onConflictDoUpdate({
+			target: gitHubIntegrations.id,
+			set: {
+				dbId: 123,
```
Comment (shige) [inline app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts]: question: Will this code be subject to changes in the future?

**Review by shige — COMMENTED** [review]
I have one question and left a comment!

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts (hunk 2 above)
Comment (toyamarinyon) [reply app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts]: Sorry! This is unnecessary code that was used during development.  I remove 634355ed850c4359ba11bdf875a7075a7d563fbb

File: app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts (hunk 1 above)
Comment (toyamarinyon) [reply app/(playground)/p/[agentId]/beta-proto/left-menu/github-integration/server-actions.ts]: Sorry! This is unnecessary code that was used during development.  I remove 634355ed850c4359ba11bdf875a7075a7d563fbb

**Review by shige — APPROVED** [review]
LGTM 👍 

I'll execute the database migrations.

**Comment by shige** [comment]
✅ 

```sh
bun drizzle-kit migrate

drizzle-kit: v0.23.0
drizzle-orm: v0.32.1

[✓] migrations applied successfully!  
```

**Pull request merged**